include/range/v3/view/exclusive_scan.hpp

```diff
@@ -0,0 +1,164 @@
+                    return ranges::begin(rng_->base());
+                }
+                range_value_type_t<Rng> read(iterator_t<Rng>) const
+                {
+                    return sum_;
+                }
+                void next(iterator_t<Rng> &it)
+                {
+                    if (it != ranges::end(rng_->base()))
+                    {
+                        auto &current = static_cast<range_value_type_t<Rng> &>(sum_);
+                        sum_ = invoke(rng_->fun_, current, *it);
+                        ++it;
+                    }
+                }
+                void prev() = delete;
+            };
+
+            adaptor<false> begin_adaptor()
+            {
+                return {*this};
+            }
+            meta::if_<use_sentinel_t, adaptor_base, adaptor<false>> end_adaptor()
+            {
+                return {*this};
+            }
+            CONCEPT_REQUIRES(Range<Rng const>() &&
+                             Invocable<Fun const&, range_common_reference_t<Rng>,
+                                     range_common_reference_t<Rng>>())
+            adaptor<true> begin_adaptor() const
+            {
+                return {*this};
+            }
+            CONCEPT_REQUIRES(Range<Rng const>() &&
+                             Invocable<Fun const&, range_common_reference_t<Rng>,
+                                     range_common_reference_t<Rng>>())
+            meta::if_<use_sentinel_t, adaptor_base, adaptor<true>> end_adaptor() const
+            {
+                return {*this};
+            }
+
+        public:
+            exclusive_scan_view() = default;
+            exclusive_scan_view(Rng rng, T init, Fun fun)
+            : exclusive_scan_view::view_adaptor{std::move(rng)}
+            , init_(std::move(init))
+            , fun_(std::move(fun))
+            {}
+            CONCEPT_REQUIRES(SizedRange<Rng>())
+            range_size_type_t<Rng> size() const
+            {
+                return ranges::size(this->base());
+            }
+        };
+
+        namespace view
+        {
+            struct exclusive_scan_fn
+            {
+            private:
+                friend view_access;
+                template<typename T, typename Fun = plus>
+                static auto bind(exclusive_scan_fn exclusive_scan, T init, Fun fun = {})
+                RANGES_DECLTYPE_AUTO_RETURN
+                (
+                        make_pipeable(std::bind(exclusive_scan, std::placeholders::_1,
+                                                std::move(init),
+                                                protect(std::move(fun))))
+                )
+            public:
+                template<typename Rng, typename T, typename Fun>
+                using Concept = meta::and_<
+                        InputRange<Rng>,
+                        MoveConstructible<T>,
+                        IndirectInvocable<Fun, iterator_t<Rng>, iterator_t<Rng>>,
```

Comment:
You also need `IndirectInvocable<Fun, iterator_t<Rng>, T*>` here if you change the "accumulator" type to `T`.